File: piotr/org.uva.sc.piotr.qls/src/main/java/qls/ast/model/Page.java
package qls.ast.model;

import qls.ast.visitors.ASTNodeVisitor;

import java.util.List;
import java.util.stream.Collectors;

public class Page extends ASTNode {

    private final String name;
    private final List<BlockElement> elements;

    public Page(MetaInformation metaInformation, String name, List<BlockElement> elements) {
        super(metaInformation);
        this.name = name;
        this.elements = elements;
    }

    @Override
    public <T> T accept(ASTNodeVisitor<T> visitor) {
        return visitor.visit(this);
    }

    public String getName() {
        return name;
    }

    public List<BlockElement> getElements() {
        return elements;
    }

    public List<DefaultDefinition> getDefaultDefinitions() {
        return this.elements.stream()
                .filter(item -> item instanceof DefaultDefinition)
                .map(item -> (DefaultDefinition) item)
                .collect(Collectors.toList())
                ;
    }
}

File: DogeQL/src/main/kotlin/qls/ast/model/Page.kt
package qls.ast.model

import qls.ast.node.QlsNode
import qls.common.Name
import qls.visitor.QlsVisitor

data class Page(val styles: List<Style>, val name: Name) : QlsNode {
    override fun <T> accept(visitor: QlsVisitor<T>): T {
        return visitor.visit(this)
    }
}